dataactvalidator/filestreaming/loadFile.py

```diff
@@ -1,53 +1,216 @@
+        print('SF133 {} {} already in database ({} records). Skipping file.'.format(
+            fiscal_year, fiscal_period, existing_records.count()))
+        return
+
+    data = pd.read_csv(filename, dtype=str)
+    data = LoaderUtils.cleanData(
+        data,
+        model,
+        {"ata": "allocation_transfer_agency",
+         "aid": "agency_identifier",
+         "availability_type_code": "availability_type_code",
+         "bpoa": "beginning_period_of_availa",
+         "epoa": "ending_period_of_availabil",
+         "main_account": "main_account_code",
+         "sub_account": "sub_account_code",
+         "fiscal_year": "fiscal_year",
+         "period": "period",
+         "line_num": "line",
+         "amount_summed":
+        "amount"},
+        {"allocation_transfer_agency": {"pad_to_length": 3},
+         "agency_identifier": {"pad_to_length": 3},
+         "main_account_code": {"pad_to_length": 4},
+         "sub_account_code": {"pad_to_length": 3},
+         "amount": {"strip_commas": True}}
+    )
+    # todo: find out how to handle dup rows (e.g., same tas/period/line number)
+    # line numbers 2002 and 2012 are the only duped line numbers,
+    # and they are not used by the validation rules, so for now
+    # just remove them before loading our SF-133 table
+    dupe_line_numbers = ['2002', '2102']
+    data = data[~data.line.isin(dupe_line_numbers)]
+
+    # add concatenated TAS field for internal use (i.e., joining to staging tables)
+    data['tas'] = data.apply(lambda row: formatInternalTas(row), axis=1)
+    # insert to db
+    LoaderUtils.insertDataframe(data, model.__table__.name, interface.engine)
+
+    # todo: insert 0 line numbers if necessary for validation rules
+
+
+def formatInternalTas(row):
+    """Concatenate TAS components into a single field for internal use."""
+    # This formatting should match formatting in dataactcore.models.stagingModels concatTas
+    tas = '{}{}{}{}{}{}{}'.format(
+        row['allocation_transfer_agency'] if row['allocation_transfer_agency'] else '000',
+        row['agency_identifier'] if row['agency_identifier'] else '000',
+        row['beginning_period_of_availa'] if row['beginning_period_of_availa'] else '0000',
+        row['ending_period_of_availabil'] if row['ending_period_of_availabil'] else '0000',
+        ' ' if pd.isnull(row['availability_type_code']) else row['availability_type_code'],
+        row['main_account_code'] if row['main_account_code'] else '0000',
+        row['sub_account_code'] if row['sub_account_code'] else '000')
+    return tas
 
 
 def loadDomainValues(basePath, localSFPath = None, localProgramActivity = None):
-    """ Load all domain value files, localSFPath is used to point to a SF-133 file, if not provided it will be downloaded from S3  """
+    """Load all domain value files, localSFPath is used to point to a SF-133 file, if not provided it will be downloaded from S3."""
```

Comment:
`localSFPath` no longer points to a file, but a directory. We should definitely update this docstring, but may also want to modify the variable name.